nibabel/freesurfer/mghformat.py

```diff
@@ -282,44 +223,71 @@ def set_data_dtype(self, datatype):
             code = self._data_type_codes[datatype]
         except KeyError:
             raise MGHError('datatype dtype "%s" not recognized' % datatype)
-        self._header_data['type'] = code
+        self._structarr['type'] = code
+
+    def _ndims(self):
+        ''' Get dimensionality of data
+
+        MGH does not encode dimensionality explicitly, so an image where the
+        fourth dimension is 1 is treated as three-dimensional.
+
+        Returns
+        -------
+        ndims : 3 or 4
+        '''
+        return 3 + (self._structarr['dims'][3] > 1)
 
     def get_zooms(self):
         ''' Get zooms from header
 
+        Returns the spacing of voxels in the x, y, and z dimensions.
+        For four-dimensional files, a fourth zoom is included, equal to the
+        repetition time (TR) in ms.
+
+        To access only the spatial zooms, use `hdr['delta']`.
+
         Returns
         -------
         z : tuple
            tuple of header zoom values
         '''
-        hdr = self._header_data
-        zooms = hdr['delta']
-        return tuple(zooms[:])
+        # Do not return time zoom (TR) if 3D image
+        tzoom = (self['tr'],)[:self._ndims() > 3]
+        return tuple(self._structarr['delta']) + tzoom
 
     def set_zooms(self, zooms):
         ''' Set zooms into header fields
 
-        See docstring for ``get_zooms`` for examples
+        Sets the spaing of voxels in the x, y, and z dimensions.
+        For four-dimensional files, a temporal zoom (repetition time, or TR, in
```

Comment:
Ouch - it's really in milliseconds?  That's unfortunate (compared to nifti).

Reply:
Annoyingly. Fortunately #567 proposes a way to get normalized zooms. (It was working on that that led to this PR in the first place.)

Reply:
Reference: https://surfer.nmr.mgh.harvard.edu/fswiki/FsTutorial/MghFormat#line-82

Reply:
Add reference to docstring?